diff --color -ru make-3.80.bak/job.c make-3.80/job.c
--- job.c	2021-01-14 10:26:42.457657979 +1100
+++ job.c	2021-01-14 10:27:03.332776801 +1100
@@ -28,8 +28,6 @@
 #include "variable.h"
 #include "debug.h"
 
-#include <string.h>
-
 /* Default shell to use.  */
 #ifdef WINDOWS32
 char *default_shell = "sh.exe";
diff --color -ru make-3.80.bak/make.h make-3.80/make.h
--- make.h	2021-01-14 10:26:42.456657974 +1100
+++ make.h	2021-01-14 10:26:51.154707480 +1100
@@ -467,9 +467,6 @@
 #if !defined (__GNU_LIBRARY__) && !defined (POSIX) && !defined (_POSIX_VERSION) && !defined(WINDOWS32)
 
 extern long int atol ();
-# ifndef VMS
-extern long int lseek ();
-# endif
 
 #endif  /* Not GNU C library or POSIX.  */
 
