diff --git perl-5.18.4/regen/regcharclass.pl perl-5.18.4/regen/regcharclass.pl
index f5cf315a54..81dcdc578d 100755
--- perl-5.18.4/regen/regcharclass.pl
+++ perl-5.18.4/regen/regcharclass.pl
@@ -1468,9 +1468,9 @@ GCB_V: Grapheme_Cluster_Break=V
 # million code points.  The results would not change unless utf8.h decides it
 # wants a maximum other than 4 bytes, or this program creates better
 # optimizations
-#UTF8_CHAR: Matches utf8 from 1 to 4 bytes
-#=> UTF8 :safe only_ascii_platform
-#0x0 - 0x1FFFFF
+UTF8_CHAR: Matches utf8 from 1 to 4 bytes
+=> UTF8 :safe only_ascii_platform
+0x0 - 0x1FFFFF
 
 # This hasn't been commented out, because we haven't an EBCDIC platform to run
 # it on, and the 3 types of EBCDIC allegedly supported by Perl would have
