--- kaem.run	2020-12-14 17:54:49.590792093 +1100
+++ kaem.run	2020-12-14 17:55:34.293053332 +1100
@@ -36,3 +36,9 @@
 ############################################
 
 ../bin/kaem --verbose --strict --file mes-m2.kaem
+
+############################################
+# Phase-14 Build blynn-compiler            #
+############################################
+
+../bin/kaem --verbose --strict --file blynn-compiler/go.kaem
