--- guix-1.5.0/m4/guix.m4
+++ guix-1.5.0/m4/guix.m4
@@ -206,9 +206,10 @@
   AC_CACHE_CHECK([whether Guile-Gcrypt is available and recent enough],
     [guix_cv_have_recent_guile_gcrypt],
     [GUILE_CHECK([retval],
-      [(use-modules (gcrypt hash))
+      [(eval-when (expand load eval)
+         (use-modules (gcrypt hash)))
        (equal? (hash-algorithm sha256)
                (lookup-hash-algorithm 'sha256))])
      if test "$retval" = 0; then
        guix_cv_have_recent_guile_gcrypt="yes"
      else
--- guix-1.5.0/configure
+++ guix-1.5.0/configure
@@ -8378,7 +8378,8 @@
 
-  $GUILE -c "(use-modules (gcrypt hash))
+  $GUILE -c "(eval-when (expand load eval)
+         (use-modules (gcrypt hash)))
        (equal? (hash-algorithm sha256)
                (lookup-hash-algorithm 'sha256))" > /dev/null 2>&1
   retval=$?
 
