--- guix-1.5.0/guix/build/gnu-build-system.scm
+++ guix-1.5.0/guix/build/gnu-build-system.scm
@@ -985,7 +985,7 @@
 
                        ;; Dump the environment variables as a shell script,
                        ;; for handy debugging.
-                       (system "export > $NIX_BUILD_TOP/environment-variables")))
+                       ))
 
                    (format #t "starting phase `~a'~%" name)
                    (with-throw-handler #t
